AppiumForMac/Server/Controller/AfMSessionController.m

```diff
@@ -574,6 +574,40 @@ -(void) closeWindow
     [self.currentWindow performAction:@"AXCancel"];
 }
 
+-(NSDictionary*)executeShellScript:(NSString*)script
+{
+    
+    NSTask *task = [[NSTask alloc] init];
+    [task setLaunchPath:@"/bin/bash"];
+    NSArray *arguments = [NSArray arrayWithObjects:
+                          @"-c" ,
+                          [NSString stringWithFormat:@"%@", script],
+                          nil];
+    
+    NSLog(@"running bash script:'%@'", script);
+    [task setArguments:arguments];
+    NSPipe *stdoutPipe = [NSPipe pipe];
+    NSPipe *stderrorPipe = [NSPipe pipe];
+    [task setStandardOutput:stdoutPipe];
+    [task setStandardError:stderrorPipe];
+    NSFileHandle *stdoutFile = [stdoutPipe fileHandleForReading];
+    NSFileHandle *stderrorFile = [stderrorPipe fileHandleForReading];
+    NSError *error;
+    if (@available(macOS 10.13, *)) {
```

Comment:
I'd rather short circuit here:

```
if (!available) {
  return;
}

...
[task launch]
```

Reply:
Ok

Reply:
Actually If I move the launch code out of that condition block Xcode complains with a warning. And I'd rather not have that. So I just trimmed the else condition block. Hope that's ok.

Reply:
stupid xcode %)